volatility3/cli/volshell/generic.py

```diff
@@ -418,67 +445,199 @@ def display_type(
+            # (layer_name) >>> dt('int')
+            # symbol_table_name1!int (4 bytes, little endian, signed)
+            data_format = volobject.vol.data_format
+            display_type_string = "{} ({} bytes, {} endian, {})".format(
+                volobject.vol.type_name,
+                data_format.length,
+                data_format.byteorder,
+                "signed" if data_format.signed else "unsigned",
+            )
+        elif hasattr(volobject.vol, "type_name"):
+            # types like void have almost no values to display other than their name, e.g.:
+            # (layer_name) >>> dt('void')
+            # symbol_table_name1!void
+            display_type_string = volobject.vol.type_name
+        else:
+            # it should not be possible to have a volobject without at least a type_name
+            raise AttributeError("Unable to find any details for object")
+
+        if include_value:  # if include_value is true also add the value to the display
+            if isinstance(volobject, objects.Pointer):
+                # for pointers include the location of the pointer and where it points to
+                return f"{display_type_string} @ {hex(volobject.vol.offset)} -> {self._display_value(volobject)}"
+            else:
+                return f"{display_type_string}: {self._display_value(volobject)}"
         else:
-            return hex(value.vol.offset)
+            return display_type_string
+
+    def _display_value(self, value: Any) -> str:
+        try:
+            # if value is a BaseAbsentValue they display N/A
+            if isinstance(value, interfaces.renderers.BaseAbsentValue):
+                return "N/A"
+            else:
+                # volobject branch
+                if isinstance(
+                    value,
+                    (interfaces.objects.ObjectInterface, interfaces.objects.Template),
+                ):
+                    if isinstance(value, objects.Pointer):
+                        # show pointers in hex to match output for struct addrs
+                        # highlight null or unreadable pointers
+                        if value == 0:
+                            suffix = " (null pointer)"
+                        elif not value.is_readable():
+                            suffix = " (unreadable pointer)"
+                        else:
+                            suffix = ""
+                        return f"{hex(value)}{suffix}"
+                    elif isinstance(value, objects.PrimitiveObject):
+                        return repr(value)
+                    elif isinstance(value, objects.Array):
+                        return repr([self._display_value(val) for val in value])
+                    else:
+                        if self.context.layers[self.current_layer].is_valid(
+                            value.vol.offset
+                        ):
+                            return f"offset: {hex(value.vol.offset)}"
```

Comment:
Think this can be `0x{value.vol.offset:x}` rather than `{hex(value.vol.offset)}`.  Absolutely a stylistic choice though...  5;)

Reply:
I don't mind either way - would you rather `0x{value.vol.offset:x}`? (can add it to the style guide PR?)

Reply:
Yeah, I think so?  Using built in format modifiers seems better than a separate function call to convert it?  Sure, I'll get it added to the coding style...

Reply:
Makes sense - done! :smile: